Queryable/Queryable/PhotoHelper/PhotoLibrary.swift

```diff
@@ -6,15 +6,14 @@ import Photos
 import os.log
 
 class PhotoLibrary {
-
-    static func checkAuthorization() async -> Bool {
+    static func checkAuthorization(status: PHAuthorizationStatus = PHPhotoLibrary.authorizationStatus(for: .readWrite)) async -> Bool {
         switch PHPhotoLibrary.authorizationStatus(for: .readWrite) {
         case .authorized:
             logger.debug("Photo library access authorized.")
             return true
         case .notDetermined:
             logger.debug("Photo library access not determined.")
-            return await PHPhotoLibrary.requestAuthorization(for: .readWrite) == .authorized
+            return await checkAuthorization(status: PHPhotoLibrary.requestAuthorization(for: .readWrite))
```

Comment:
Okay, I think this fixes the issue! I was only considering limited authorization when checking if _already_ authorized. I also needed to accept limited authorization at the time of request.

Let me know if you want me to restructure this code somehow. I think the recursive nature is a bit weird, but I assume the OS won't return `notDetermined` repeatedly. (Maybe they'll add a cancel button in the future that changes that though?)

It's worth noting there's a minor logging behavior change here. Previously, you wouldn't log the new status after requesting authorization. Now, we do.

If you want me to refactor this, let me know how! If logging new status after request isn't your preference, I can just do something simple like `[.authorized, .limited].contains(PHPhotoLibrary.requestAuthorization(for: .readWrite))`.